# Snapshot report for `packages/build/tests/error/plugin_load/tests.js`

The actual snapshot is saved in `tests.js.snap`.

Generated by [AVA](https://ava.li).

## Early exit

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
       - /file/path from netlify.toml␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └─────────────────────────────────────┘␊
    ␊
    onPreBuild␊
    ␊
    (/file/path onPreBuild completed in 1ms)␊
    ␊
    ┌───────────────────────────────────────┐␊
    │ 2. onBuild command from /file/path │␊
    └───────────────────────────────────────┘␊
    ␊
    onBuild start␊
    onBuild end␊
    ␊
    (/file/path onBuild completed in 1ms)␊
    ␊
    ┌──────────────────────────────────────┐␊
    │ 3. onPostBuild command from /file/path │␊
    └──────────────────────────────────────┘␊
    ␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      Could not receive event from child process because it already exited.␊
      The plugin might have exited due to a bug terminating the process, such as an infinite loop.␊
      The plugin might also have explicitly terminated the process, for example with process.exit().␊
      Plugin methods should instead:␊
        - on success: return␊
        - on failure: call utils.build.failPlugin() or utils.build.failBuild()␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      In "onPostBuild" event in "/file/path" from netlify.toml␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## Exits in plugins

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └─────────────────────────────────────┘␊
    ␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      Plugin exited with exit code 1 and signal null.␊
      The plugin might have exited due to a bug terminating the process, such as an infinite loop.␊
      The plugin might also have explicitly terminated the process, for example with process.exit().␊
      Plugin methods should instead:␊
        - on success: return␊
        - on failure: call utils.build.failPlugin() or utils.build.failBuild()␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      In "onPreBuild" event in "/file/path" from netlify.toml␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## No repository root

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌────────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └────────────────────────────────────────┘␊
    ␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └─────────────────────────────────────┘␊
    ␊
      Error message␊
      Error: test␊
    ␊
      Error location␊
      In "onPreBuild" event in "/file/path" from netlify.toml␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## Node module all fields

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: netlify-plugin-test␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - netlify-plugin-test@1.0.0 from netlify.toml and package.json␊
    ␊
    ┌────────────────────────────────────────────────┐␊
    │ 1. onPreBuild command from netlify-plugin-test │␊
    └────────────────────────────────────────────────┘␊
    ␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ Plugin "netlify-plugin-test" failed │␊
    └─────────────────────────────────────┘␊
    ␊
      Error message␊
      Error: test␊
    ␊
      Plugin details␊
      Package:        netlify-plugin-test␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/test/test.git␊
      npm link:       https://www.npmjs.com/package/netlify-plugin-test␊
      Report issues:  https://test.com/test␊
    ␊
      Error location␊
      In "onPreBuild" event in "netlify-plugin-test" from netlify.toml and package.json␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: netlify-plugin-test`

## Node module partial fields

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: netlify-plugin-test␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - netlify-plugin-test@ from netlify.toml and package.json␊
    ␊
    ┌────────────────────────────────────────────────┐␊
    │ 1. onPreBuild command from netlify-plugin-test │␊
    └────────────────────────────────────────────────┘␊
    ␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ Plugin "netlify-plugin-test" failed │␊
    └─────────────────────────────────────┘␊
    ␊
      Error message␊
      Error: test␊
    ␊
      Plugin details␊
      Package:        netlify-plugin-test␊
    ␊
      Error location␊
      In "onPreBuild" event in "netlify-plugin-test" from netlify.toml and package.json␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: netlify-plugin-test`

## Process warnings

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └─────────────────────────────────────┘␊
    ␊
    onPreBuild␊
    ␊
    (/file/path onPreBuild completed in 1ms)␊
    ␊
    ┌─────────────────────────────┐␊
    │   Netlify Build Complete    │␊
    └─────────────────────────────┘␊
    ␊
    (Netlify Build completed in 1ms)␊
    ␊
     ‼  warning (Warning)  test␊
    STACK TRACE␊
    STACK TRACE`

## Top function errors local

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      Error: Could not load plugin:␊
      test␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      While loading "/file/path" from netlify.toml␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## Top-level errors

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      Error: Could not import plugin:␊
      test␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      While loading "/file/path" from netlify.toml␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## Uncaught exception

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └─────────────────────────────────────┘␊
    ␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      UncaughtException: an exception was thrown but not caught: Error: test␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      In "onPreBuild" event in "/file/path" from netlify.toml␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`

## Unhandled promises

> Snapshot 1

    `␊
    ┌─────────────────────────────┐␊
    │        Netlify Build        │␊
    └─────────────────────────────┘␊
    ␊
    > Version␊
      @netlify/build 1.0.0␊
    ␊
    > Flags␊
      repositoryRoot: /file/path␊
    ␊
    > Current directory␊
      /file/path␊
    ␊
    > Config file␊
      /file/path␊
    ␊
    > Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path␊
    ␊
    > Context␊
      production␊
    ␊
    > Loading plugins␊
       - /file/path from netlify.toml␊
    ␊
    ┌─────────────────────────────────────┐␊
    │ 1. onPreBuild command from /file/path │␊
    └─────────────────────────────────────┘␊
    ␊
    onPreBuild␊
    ␊
    ┌──────────────────────────────────┐␊
    │ Plugin "/file/path" internal error │␊
    └──────────────────────────────────┘␊
    ␊
      Error message␊
      UnhandledRejection: a promise was rejected but not handled: Error: test␊
    ␊
      Plugin details␊
      Package:        /file/path␊
      Version:        1.0.0␊
      Repository:     git+https://github.com/netlify/build.git␊
      npm link:       https://www.npmjs.com/package/@netlify/build␊
      Report issues:  https://github.com/netlify/build/issues␊
    ␊
      Error location␊
      In "onPreBuild" event in "/file/path" from netlify.toml␊
    STACK TRACE␊
    ␊
      Resolved config␊
      plugins:␊
        - inputs: {}␊
          origin: config␊
          package: /file/path`
